scaladoc/src/dotty/tools/scaladoc/snippets/SnippetChecker.scala

```diff
@@ -0,0 +1,60 @@
+package dotty.tools.scaladoc
+package snippets
+
+import dotty.tools.scaladoc.DocContext
+import java.nio.file.Paths
+import java.io.File
+
+import dotty.tools.io.AbstractFile
+import dotty.tools.dotc.fromtasty.TastyFileUtil
+import dotty.tools.dotc.config.Settings._
+import dotty.tools.dotc.config.ScalaSettings
+
+class SnippetChecker(val args: Scaladoc.Args)(using cctx: CompilerContext):
+
+// (val classpath: String, val bootclasspath: String, val tastyFiles: Seq[File], isScalajs: Boolean, useJavaCp: Boolean):
+  private val sep = System.getProperty("path.separator")
+
+  private val fullClasspath = List(
+    args.tastyFiles
+      .map(_.getAbsolutePath())
+      .map(AbstractFile.getFile(_))
+      .flatMap(t => try { TastyFileUtil.getClassPath(t) } catch { case e: AssertionError => Seq() })
+      .distinct.mkString(sep),
+    args.classpath
+  ).mkString(sep)
+
+  private val snippetCompilerSettings: Seq[SnippetCompilerSetting[_]] = cctx.settings.userSetSettings(cctx.settingsState).filter(_ != cctx.settings.classpath).map( s =>
+    SnippetCompilerSetting(s, s.valueIn(cctx.settingsState))
+  ) :+ SnippetCompilerSetting(cctx.settings.classpath, fullClasspath)
```

Comment:
What is the purpose of SnippetCompilerSetting? If it's just storing settings it seems that the existing `SettingsState` would suffice and you could write something like :
```scala
  private val snippetCompilerSettings = cctx.settings.classpath.updateIn(cctx.settingsState, fullClasspath)
```

Reply:
If we change classpath of compiler context used to load TASTY to produce documentation from, won't it affect loading process?

Reply:
Since updateIn returns a new SettingsState, I was expecting it to not mutate the existing settingsState of the compiler context, but in fact it's slightly more tricky than that, it will be mutated if no one tried to read a setting previously: https://github.com/lampepfl/dotty/blob/e2e77b56d012b00faf16a86ad1c8437404bb0812/compiler/src/dotty/tools/dotc/config/Settings.scala#L36-L40 (this is a really bad idea and that code that should be replaced)